### PR TITLE
Logs CFN errors and success from feature branch builds

### LINKED ISSUE #43: Add logic to deployFeatureBranch to retry if deploy fails

### LINKED ISSUE #115: Send cfn log messages to slack when deployments fail in circle
https://github.com/candrholdings/slack-cli

### PR DESCRIPTION
**Pull request by thebho**
## Description
After a CFN deploy this will pull the stack events are parse the events for errors/successes from the previous deployment
If the deployment failed and it needs to be deleted, it will call a delete stack on that stack

## Motivation and Context
Fixes #43 
Fixes #115 
Tested with new and updated builds
Tested with broken templates

## Screenshots (if appropriate):

### CODE REVIEW

**Comment by thebho** [comment]
This could be used for master builds in the future, but would have to remove the delete option

File: scripts/cfn.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import boto3
+from boto3.s3.transfer import ClientError
+import os
+import json
+import sys
+
+def main(arg1):
```
Comment (dane-johnson) [inline scripts/cfn.py]: Variable name could be more descriptive

File: scripts/cfn.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    for event in response['StackEvents']:
+        # The beginning of the current Stack Event
+        if event['ResourceType'] == 'AWS::CloudFormation::Stack':
+            if event['ResourceStatus']  ==  'REVIEW_IN_PROGRESS' or event['ResourceStatus']  ==  'UPDATE_IN_PROGRESS':
+                print 'Stack Deploy Failed'
+                if rollback:
+                    print 'Deleting Stack'
+                    client.delete_stack(
+                        StackName=arg1
+                )
```
Comment (dane-johnson) [inline scripts/cfn.py]: Looks like indentation is off here.

File: scripts/cfn.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import boto3
+from boto3.s3.transfer import ClientError
+import os
+import json
+import sys
+
+def main(arg1):
+    dateFormat = "[%H:%M %m.%d.%y]"
+    rollback = False
```
Comment (dane-johnson) [inline scripts/cfn.py]: If this is a global constant, it shouldn't have local scope

File: scripts/cfn.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                    client.delete_stack(
+                        StackName=arg1
+                )
+                return
+        # Delet Stacks that were Rolled Back
+        if event['ResourceStatus'] == 'ROLLBACK_COMPLETE':
+            rollback = True
+        # Catch and log stack failures
+        if event['ResourceStatus'] == 'CREATE_FAILED' or event['ResourceStatus']  == 'UPDATE_FAILED':
+            print 'Stack Event failed ' + event['Timestamp'].strftime(dateFormat)
```
Comment (dane-johnson) [inline scripts/cfn.py]: Since this is done many times, it can be made into a helper function

**Pull request merged**